android/app/src/main/java/com/microsoft/codepush/react/CodePush.java

```diff
@@ -119,15 +128,22 @@ public String getBundleUrl(String assetsBundleFileName) {
                 return binaryJsBundleUrl;
             }
 
-            File packageFile = new File(packageFilePath);
-            if (packageFile.lastModified() < binaryResourcesModifiedTime) {
+            ReadableMap packageMetadata = codePushPackage.getCurrentPackage();
+            // May throw NumberFormatException.
+            Long binaryModifiedDateDuringPackageInstall = Long.parseLong(CodePushUtils.tryGetString(packageMetadata, BINARY_MODIFIED_TIME_KEY));
+            if (binaryModifiedDateDuringPackageInstall == binaryResourcesModifiedTime) {
+                return packageFilePath;
+            } else {
                 // The binary version is newer.
+                CodePushUtils.log("Found a package installed via CodePush that was installed " +
```

Comment:
What if we simply said "Loading JS bundle from binary", and then add another log to the CodePush path that said "Loading JS bundle from latest CodePush update"?